irmaclient/logs.go

```diff
@@ -35,6 +36,72 @@ type LogEntry struct {
 	request    irma.SessionRequest   // cached parsed version of Request; get with LogEntry.SessionRequest()
 }
 
+// We need manual unmarshalling to deal with legacy log entries that have
+func (entry *LogEntry) UnmarshalJSON(b []byte) (err error) {
+	// Internal type needed for unmarshalling to delay requestorInfo unmarshalling
+	// NOTE: keep this up to date with above definition of log entry!
+	var temp struct {
+		// General info
+		ID   uint64
+		Type irma.Action
```

Comment:
Instead of this copy of the LogEntry struct that needs to be kept in sync, you can also create a new type with the same underlying type as `LogEntry` by saying
```go
type rawLogEntry LogEntry
var temp rawLogEntry
```

Reply:
That is not sufficient in this case because I want to delay unmarshaling of the ServerName field, and thus need it to have a different type.

Reply:
An option would be to make a type without `ServerName` in it (lets use @sietseringers name `rawLogEntry`) and then make LogEntry `type struct { rawLogEntry, ServerName ... }` by including the other struct. Then you can do the same in your temp struct and include another type for `ServerName`

Reply:
Found a better aproach that uses shadowing